external_search.py

```diff
@@ -48,99 +60,189 @@ def __init__(self, url=None, works_index=None):
+        index_details = self.indices.get_alias(name=current_alias, ignore=[404])
+        found = not (index_details.get('status')==404 or 'error' in index_details)
+
+        def _set_works_index(name):
+            self.works_index = self.__client.works_index = name
+
+        if found:
+            # We found an index for the alias in configuration. Assume
+            # there is only one.
+            _set_works_index(index_details.keys()[0])
+
+        if current_alias.endswith(self.CURRENT_ALIAS_SUFFIX):
+            # The alias culled from configuration is intended to be
+            # a current alias. Get the index name from here.
+            base_works_index = self._base_works_index(current_alias)
+            version = ExternalSearchIndexVersions.latest()
+            _set_works_index(base_works_index+'-'+version)
+        else:
+            # Without the CURRENT_ALIAS_SUFFIX, assume the index string
+            # from config is the index itself and needs to be swapped.
+            _set_works_index(current_alias)
+
         if not self.indices.exists(self.works_index):
             self.setup_index()
+        self.setup_current_alias()
 
-    def setup_index(self):
-        """
-        Create the search index with appropriate mapping.
+    def setup_index(self, new_index=None):
+        """Create the search index with appropriate mapping.
 
         This will destroy the search index, and all works will need
         to be indexed again. In production, don't use this on an
         existing index. Use it to create a new index, then change the 
         alias to point to the new index.
         """
-
-        if self.works_index:
-            if self.indices.exists(self.works_index):
-                self.indices.delete(self.works_index)
-
-            self.log.info("Creating index %s", self.works_index)
-            self.indices.create(
-                index=self.works_index,
-                body={
-                    "settings": {
-                        "analysis": {
-                            "filter": {
-                                "en_stop_filter": {
-                                    "type": "stop",
-                                    "stopwords": ["_english_"]
-                                },
-                                "en_stem_filter": {
-                                    "type": "stemmer",
-                                    "name": "english"
-                                },
-                                "en_stem_minimal_filter": {
-                                    "type": "stemmer",
-                                    "name": "english"
-                                },
+        index = new_index or self.works_index
+
+        if self.indices.exists(index):
+            self.indices.delete(index)
+
+        self.log.info("Creating index %s", index)
+        self.indices.create(
+            index=index,
+            body={
```

Comment:
The body should be tied to the version number as well as the mapping - the analyzers are one of the more likely things to change between versions.